### PR TITLE
feat: adds `customGroup.newlinesInside ` option

### PR DESCRIPTION
**Pull request by hugop95**
First PR aiming to partially resolve the issue https://github.com/azat-io/eslint-plugin-perfectionist/issues/401.
- ⚙️ **PR 1 (this)**: Add `customGroup.newlinesInside` option. We will take the opportunity to create a `getNewlinesBetweenOption` method that will help us later to precisely customize newline behavior between two specific groups. 
- ❌ **PR 2**: Add a way to choose a specific newline behavior between two groups. See the proposals [here](https://github.com/azat-io/eslint-plugin-perfectionist/issues/401#issuecomment-2558581331).

# Description

We currently have the `newlinesBetween` option, which applies the same newlines-related rules between all groups.
This PR adds a sub-option for the new array-based `customGroups`:
- `newlinesInside?: 'always' | 'never'`

Here is the list of rules with this new addition:
- `sort-classes`
- `sort-modules`
- `sort-object-types`
- `sort-interfaces`

# New option

## `customGroup.newlinesInside?: 'ignore' | 'always' | 'never'`

Newlines rule that should be applied between two consecutive nodes of the **same** custom group.

Example: Enforce a newline between all class methods.
```ts
{
  customGroups: [
    {
      groupName: "methodsWithNewlinesBetween",
      selector: "method",
      newlinesInside: "always"
    }
  ]
}
```

# Other changes

- Unifies the way spacing errors are reported:
  - The `leftNum < rightNum` check is currently only made if `numberOfEmptyLinesBetween === 0` (as confirmed by passing tests).
  - With `newlinesBetween: always` and if `numberOfEmptyLinesBetween > 1`, we were not checking if `leftNum < rightNum`.
  - The `leftNum < rightNum` check is not made when `newlinesBetween: never` either (as confirmed by passing tests).
  - => Always check if `leftNum < rightNum`.
  - The few tests impacted can be found in those commits: 04132ee4bd1bbbebf2b0c27f802cf77e8aa829ea, 252873c1a821ce92ae7307396d499f6b8df29452.
    - Only some spacing errors reported were removed, outputs are unchanged.

### What is the purpose of this pull request?

- [x] New Feature

### CODE REVIEW

**Comment by hugop95** [comment]
@OlivierZal 👋 Feel free to review this if you want 🙂

File: test/get-newlines-between-option.test.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
+import { describe, expect, it } from 'vitest'
```
Comment (azat-io) [inline test/get-newlines-between-option.test.ts:1]: Can you move this file to `test/utils/`?

Comment (hugop95) [reply test/get-newlines-between-option.test.ts]: @azat-io Oops, rebase from https://github.com/azat-io/eslint-plugin-perfectionist/pull/425 didn't move it 😅. Moved in f4b4ad5060d0c87da7621928c4d73e5ac8c53644.

**Review by OlivierZal — APPROVED** [review]

File: test/get-newlines-between-option.test.ts (hunk 1 above)
Comment (OlivierZal) [reply test/get-newlines-between-option.test.ts]: Looks good to me!

File: test/rules/sort-imports.test.ts [hunk 2]
```diff
@@ -290,13 +290,6 @@ describe(ruleName, () => {
               },
               messageId: 'unexpectedImportsGroupOrder',
             },
-            {
```
Comment (azat-io) [inline test/rules/sort-imports.test.ts:293]: Can you tell me why you removed these tests?

**Review by azat-io — APPROVED** [review]

**Pull request merged**